WcaOnRails/app/webpacker/javascript/hooks/useInputState.js

```diff
@@ -0,0 +1,24 @@
+import { useState, useCallback } from 'react';
+
+// All the react-semantic-ui inputs expect an 'onChange' handler with the signature:
+// onChange(event: ChangeEvent, data: object)
+// It's tempting to use 'event.target.value', which works in most cases
+// but not for Form.Select for instance (!!), so we better use 'data.value'
+// which always holds what we want.
+// This is a small wrapper to automatically create the appropriate callback
+// and avoid rendering the input at each render.
+// If 'data' is undefined, we consider it's a regular "setState" call and set
+// the value from the first param.
+const useInputState = (defaultVal = undefined) => {
+  const [state, setState] = useState(defaultVal);
+  const updateFromOnChange = useCallback((ev, data = undefined) => {
+    if (data) {
+      setState(data.value);
+    } else {
+      setState(ev);
```

Comment:
If this is a DOM event we may want `event.target.value` (but sometimes `event.target.checked`). Is the assumption that in those cases `data` is always given and has proper value? I can imagine someone passing the callback to regular input, in which case this wouldn't work.

Reply:
This may or may not be a dom event: the purpose is to have the `setState` used in two ways:
  - from the react-semantic-ui input, in which case `ev` is a dom event but react-semantic-ui guarantees that `data.value` always has the value we want so we should just use it.
  - directly from the user, in a similar fashion as the regular `setState`, in which case `data` is undefined, and `ev` actually contains the new state value.

The following extract from [this component](https://github.com/thewca/worldcubeassociation.org/pull/6360/files#diff-bbf54e09245126ac14da42de3ae29be6aae5b23cfed7db05753552a8338933a3) showcases the advantage of this approach:
```js
// Create the state
const [wcaId, setWcaId] = useInputState('');

// Use it like the regular setState
useEffect(() => {
  setWcaId('');
}, [setWcaId, name]);

// Or pass it done a Form.Input
<Form.Input
  label="WCA ID"
  onChange={setWcaId}
  error={errors.wca_id}
  value={wcaId}
/>
```

> I can imagine someone passing the callback to regular input, in which case this wouldn't work.

This absolutely wouldn't work indeed (and it's not intended for use with regular inputs)!
I think the small documentation text I wrote gives this away, but just to be sure I've added an explicit warning about that!